common/export-common-cartridge-util/src/java/org/sakaiproject/exporter/util/CCExport.java

```diff
@@ -491,6 +495,24 @@ private StringBuilder convertSlashToHyphen(StringBuilder parsedContent, Map<Stri
 		return parsedContent;
 	}
 
+	private StringBuilder addPDFViewer(StringBuilder parsedContent, Map<String, Boolean> savedLinks) {
+		// Add the CSS class and URL parameters so that Canvas displays the 'Preview the file' icon next to the filename.
+		for (Map.Entry<String, Boolean> linkPath : savedLinks.entrySet()) {
+			if (linkPath.getKey().toLowerCase().contains(".pdf")) {
+
+				String pat = "(href=)+(\"|')?(/group/" + siteId + "/)(" + linkPath.getKey() + ")(\"|')?([ a-z0-9=\"'-]*)(>)";
+				Pattern target = Pattern.compile(pat, Pattern.CASE_INSENSITIVE);
+				Matcher matcher = target.matcher(parsedContent);
+
+				if (matcher.find()) {
```

Comment:
Should this be in a loop so that if there are multiple links to the same file they all get updated?

Reply:
Loop added and region reset so that the extra content added means the whole string is parsed.